cEP-0019.md

```diff
@@ -0,0 +1,404 @@
+# Meta-review System
+
+| Metadata |                                            |
+| -------- | ------------------------------------------ |
+| cEP      | 0019                                       |
+| Version  | 1.0                                        |
+| Title    | Meta-review System                         |
+| Authors  | Boxuan Li <mailto:liboxuan@connect.hku.hk> |
+| Status   | Proposed                                   |
+| Type     | Process                                    |
+
+## Abstract
+
+This cEP describes the details of the process of implementing the meta-review
+system as a part of the
+[GSoC'18 project](https://summerofcode.withgoogle.com/projects/#5188493739819008).
+
+## Background
+
+People respond to review comments on pull requests by giving feedback. Emojis
+make it much easier to give and receive feedback without a long comment thread.
+Those emojis are called
+[reactions](https://developer.github.com/v4/enum/reactioncontent/) on GitHub
+or [award emoji](https://docs.gitlab.com/ee/user/award_emojis.html) on GitLab.
+THUMBS_UP and THUMBS_DOWN, together with other reactions, are used by people
```

Comment:
There are not many reactions on GitHub.  Better to list them all, after you've explained that  GitLab is out of scope.

Also, later in this document, we should give a 'meta-review' meaning to each of them.

Reply:
Remove `THUMBS_UP and THUMBS_DOWN, together with other reactions...` from here.

Explain background on how they are *currently* used after providing the list.

Do not guess how they are being used in "Background".  You already say at the bottom of this section that they "are not collected and analyzed", so we dont currently know how they are being used.  We might say current usage is adhoc.

Ultimately, after this cEP is approved, how they were being used becomes irrelevant.  After this cEP is merged, this cEP will dictate how they will be used by the coala community members in the future.

And this cEP will provide tools for analysing how they will be used in the future.

And this cEP should probably also consider how to track incorrect usage of them.... ;-)

Luckily reactions can be modified at any time, which means incorrect usage (e.g. by newcomers not yet aware of this cEP) can be rectified.

But that also adds new challenges this cEP should address - what is someone maliciously decides to remove all of their positive feedback of other community members, e.g. to try to influence gsoc allocation.